src/basic/src/ScalarFunction.C

```diff
@@ -92,8 +111,8 @@ BaseScalarFunction<V, M>::lnValue(const V & domainVector, V & gradVector) const
     // Store the old value of the perturbed element so we can undo it later
     double tmp = perturbedVector[i];
 
-    perturbedVector[i] += h;
-    gradVector[i] = (this->lnValue(perturbedVector) - value) / h;
+    perturbedVector[i] += m_fdStepSize;
+    gradVector[i] = (this->lnValue(perturbedVector) - value) / m_fdStepSize;
 
```

Comment:
You can get a little more flexibility in most cases by doing central rather than one-sided differences.  2N instead of N evaluations, but epsilon^2 instead of epsilon error.

On the other hand, there are a lot of parameters where the domain is "non-negative numbers" and where the evaluation point is "just barely non-negative", and there is something to be said for only doing perturbations in the positive direction.